src/libsyntax/attr.rs

```diff
@@ -317,6 +317,13 @@ pub fn find_export_name_attr(diag: &Handler, attrs: &[Attribute]) -> Option<Inte
     })
 }
 
+pub fn contains_extern_indicator(attrs: &[Attribute]) -> bool {
+    contains_name(attrs, "no_mangle") ||
+        contains_name(attrs, "link_section") ||
+        contains_name(attrs, "linkage") ||
```

Comment:
`link_section` I'd probably agree to remove, but `linkage` certainly is relevant. As #29676 checked, `extern` and `extern_weak` are the values that cause a symbol to be reachable, so this is slightly broad in that it'll tag `linkage="internal"` items, but doing so is more logic and string literals so not sure which way to go with it?